src/deepsparse/loggers/helpers.py

```diff
@@ -34,11 +34,60 @@
     "NO_MATCH",
     "access_nested_value",
     "finalize_identifier",
+    "unwrap_logs_dictionary",
 ]
 
 NO_MATCH = "NO_MATCH"
 
 
+def unwrap_logs_dictionary(
+    value: Any, parent_identifier="", seperator="__"
+) -> Generator[Tuple[str, Any], None, None]:
+    """
+    Unwrap the `value`, given that it may be nested dictionary.
+    e.g.
+    ```
+    value = {"foo": {"alice": 1, "bob": 2},
+             "bazz": 2},
+    for identifier, value in unwrap_possible_dictionary(value):
+        -> yields:
+            "foo__alice", 1
+            "foo__bob", 2
+            "bazz", 2 (no unwrapping)
+    ```
+
+    :param value: The value to possibly unwrap
+    :param parent_identifier: The identifier that may be prepended to the
+        child identifier retrieved from the nested dictionary
+    :param seperator: The seperator to use when composing the parent and child
+        identifiers
+    :return: A generator that:
+        - if `value` is not a dictionary:
+            yields the `parent_identifier` and `value`
+        - else:
+            yields a sequence of tuples (identifier, value) where:
+                identifier is composed by connecting the keys over the multiple levels
+                    of nesting with the seperator
+                value is extracted from the nested dictionary (corresponding to the
+                    appropriate composed identifier)
+    """
+    if not isinstance(value, dict):
+        yield parent_identifier, value
+    else:
+        for child_identifier, child_value in value.items():
```

Comment:
good job with the variable names, very easy to track

Reply:
Thanks!